nsxt/resource_nsxt_policy_domain.go

```diff
@@ -0,0 +1,336 @@
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+// This resource is supported only for Policy Global Manager
+func resourceNsxtPolicyDomain() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyDomainCreate,
+		Read:   resourceNsxtPolicyDomainRead,
+		Update: resourceNsxtPolicyDomainUpdate,
+		Delete: resourceNsxtPolicyDomainDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"location": {
+				Type:        schema.TypeSet,
+				Description: "Locations where this domain is deployed",
+				Required:    true,
+				MinItems:    1,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceNsxtPolicyDomainExists(id string, connector *client.RestConnector, isGlobalManager bool) (bool, error) {
+	var err error
+	if isGlobalManager {
+		client := gm_infra.NewDefaultDomainsClient(connector)
+		_, err = client.Get(id)
+	} else {
+		return false, fmt.Errorf("Domain resource is not supported for local manager")
+	}
+
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	return false, logAPIError("Error retrieving domain", err)
+}
+
+func createChildDomainDeploymentMap(m interface{}, domainID string, location string) (*data.StructValue, error) {
+	converter := bindings.NewTypeConverter()
+	converter.SetMode(bindings.REST)
+
+	mapID := domainID + "-" + location
+	path := getGlobalPolicyEnforcementPointPathWithLocation(m, location)
+
+	Type := "DomainDeploymentMap"
+	mapObj := model.DomainDeploymentMap{
+		Id:                   &mapID,
+		DisplayName:          &location,
+		EnforcementPointPath: &path,
+		ResourceType:         &Type,
+	}
+
+	childMap := model.ChildDomainDeploymentMap{
+		Id:                  &mapID,
+		ResourceType:        "ChildDomainDeploymentMap",
+		DomainDeploymentMap: &mapObj,
+	}
+
+	dataValue, errors := converter.ConvertToVapi(childMap, model.ChildDomainDeploymentMapBindingType())
+	if len(errors) > 0 {
+		return nil, errors[0]
+	}
+	return dataValue.(*data.StructValue), nil
+}
+
+func setDomainStructWithChildren(m interface{}, domain *model.Domain, locations []string, isUpdate bool) error {
+	var children []*data.StructValue
+
+	// Loop over locations
+	for _, location := range locations {
+		child, err := createChildDomainDeploymentMap(m, *domain.Id, location)
+		if err != nil {
+			return err
+		}
+		if child != nil {
+			children = append(children, child)
+		}
+	}
+
+	if isUpdate {
+		// Delete old existing locations
+		connector := getPolicyConnector(m)
+		converter := bindings.NewTypeConverter()
+		dmClient := gm_domain.NewDefaultDomainDeploymentMapsClient(connector)
```

Comment:
Please consider, if not yet, getting old locations from the state without backend call, using d.GetChange("location")

Reply:
Thanks. I will try that

Reply:
Actually I don't think it will help much, because I also need the ID of the one to be deleted, and I don't have it in the scheme. It could be an imported resources so the ID can be anyhting

Reply:
yes, good point